### PR TITLE
CNF-10430: gather and recover lvm devices 

### PR DESCRIPTION
**Pull request by Missxiaoguo**

### CODE REVIEW

File: ibu-imager/postpivot/postpivot.go [hunk 1]
```diff
@@ -246,3 +252,22 @@ func (p *PostPivot) applyManifests() error {
 	}
 	return nil
 }
+
+func (p *PostPivot) recoverLvmDevices() error {
+	lvmConfigPath := path.Join(p.workingDir, common.LvmConfigDir)
```
Comment (tsorya) [inline ibu-imager/postpivot/postpivot.go:257]: can it be run couple of times? If not please use utils.runonce as service can be restarted

File: ibu-imager/postpivot/postpivot.go [hunk 2]
```diff
@@ -246,3 +252,22 @@ func (p *PostPivot) applyManifests() error {
...
+	lvmDevicesPath := path.Join(lvmConfigPath, path.Base(common.LvmDevicesPath))
+
+	p.log.Infof("Recovering lvm devices from %s", lvmDevicesPath)
+	if err := cp.Copy(lvmDevicesPath, common.LvmDevicesPath); err != nil {
+		if !os.IsNotExist(err) {
```
Comment (tsorya) [inline ibu-imager/postpivot/postpivot.go:262]: Error only on not exists?

File: internal/clusterconfig/lvmconfig_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			expectedNumOfFiles: 0,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			tmpDir := t.TempDir()
+
+			// Create the original files
+			hostDir = filepath.Join(tmpDir, "/host")
```
Comment (leo8a) [inline internal/clusterconfig/lvmconfig_test.go:56]: ?
```suggestion
			hostDir = filepath.Join(tmpDir, "host")
```

File: ibu-imager/postpivot/postpivot.go [hunk 4]
```diff
@@ -246,3 +252,22 @@ func (p *PostPivot) applyManifests() error {
...
+			return err
+		}
+	}
+
+	// Update the online record of PVs and activate all lvm devices in the VGs
+	_, err := p.ops.RunInHostNamespace("pvscan", "--cache", "--activate", "ay")
+	if err != nil {
+		return fmt.Errorf("Failed to scan and active lvm devices, err: %w", err)
```
Comment (leo8a) [inline ibu-imager/postpivot/postpivot.go:270]: s/Failed/failed/

**Pull request merged**